specs/sharding/beacon-chain.md

```diff
@@ -697,7 +707,8 @@ def process_shard_header(state: BeaconState, signed_header: SignedShardBlobHeade
 
     # Verify that the block root matches,
     # to ensure the header will only be included in this specific Beacon Chain sub-tree.
-    assert header.body_summary.beacon_block_root == get_block_root_at_slot(state, slot - 1)
+    assert header.body_summary.beacon_block_root in 
```

Comment:
Maybe instead we just specify the slot the root refers to with an additional field? Also note that for each gap slot, the root is the same

Reply:
Is it about optimization? 
I could suggest `compute_previous_slots` to return slots in reverse order. So in vast majority of cases that would be an equivalent to the original check complexity